src/iface/ethernet.rs

```diff
@@ -951,9 +951,15 @@ impl<'b, 'c, 'e> InterfaceInner<'b, 'c, 'e> {
     /// after a query is broadcasted by a router.
     #[cfg(feature = "proto-igmp")]
     fn process_igmp<'frame>(&mut self, timestamp: Instant, ipv4_repr: Ipv4Repr, ip_payload: &'frame [u8]) ->
-                             Result<Packet<'frame>> {
-        let igmp_packet = IgmpPacket::new_checked(ip_payload)?;
-        let igmp_repr = IgmpRepr::parse(&igmp_packet)?;
+                           Result<Packet<'frame>> {
+        let igmp_packet = match IgmpPacket::new_checked(ip_payload) {
+            Ok(igmp_packet) => igmp_packet,
+            Err(_) => return Ok(Packet::None),
+        };
+        let igmp_repr = match IgmpRepr::parse(&igmp_packet) {
+            Ok(igmp_repr) => igmp_repr,
+            Err(_) => return Ok(Packet::None),
+        };
```

Comment:
Why did you do this? smoltcp intentionally pops up errors to the very top of the stack, such that if you receive malformed packets and nothing works, you do not have to wonder as to what is the cause.

Reply:
You said you want the example to panic on Err. That was happening to me all the time due to IGMPv3 traffic and invalid checksums.

Reply:
Thanks for the explanation. I will revert this part, and make sure examples log errors instead of crashing on them. I did not realize that IGMPv3 traffic would cause this.